x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_hosts_url_state.ts

```diff
@@ -55,14 +56,16 @@ const reducer = (state: HostsState, action: Action): HostsState => {
     case 'setFilter':
       return { ...state, filters: [...action.payload] };
     case 'setQuery':
-      const { filters, query, ...payload } = action.payload;
+      const { filters, query, panelFilters, ...payload } = action.payload;
       const newFilters = !filters ? state.filters : filters;
+      const newControlPanelFilters = !panelFilters ? state.panelFilters : panelFilters;
```

Comment:
Storing the panelFilters here allows as to trigger the snapshot request earlier than waiting for the control panels to load.